### ISSUE TITLE
`docker_additional_options` is not passed to container when building with poetry

### ISSUE THREAD

**Issue by FurqanHabibi** [terraform-aws-modules/terraform-aws-lambda#427]
## Description

We are trying to pass the following environment variable to the docker container via `docker_additional_options`:
```hcl
docker_additional_options = ["-e", "PIP_PLATFORM=manylinux2014_aarch64"]
```
but as can be seen in the following output, said options is not being passed to the docker run command:
```
module.le_auto_stack.module.lambda_issue_layer.null_resource.archive[0] (local-exec): > docker run --rm -w /var/task -v /tmp/terraform-aws-lambda-fyvlzv2k:/var/task:z -v /home/runner/.ssh/known_hosts:/root/.ssh/known_hosts:z --entrypoint '' 1d556381b475 /bin/sh -c 'poetry config --no-interaction virtualenvs.create true && poetry config --no-interaction virtualenvs.in-project true && poetry export --format requirements.txt --output requirements.txt --with-credentials && python3.9 -m pip install --no-compile --no-deps --prefix= --target=. --requirement=requirements.txt && chown -R 1001:123 .'
```

- [x] ✋ I have searched the open/closed issues and my issue is not listed.

## Versions

- Module version: 4.10.1
- Terraform version:v1.3.7
- Provider version(s):
```
+ provider registry.terraform.io/hashicorp/aws v4.56.0
+ provider registry.terraform.io/hashicorp/external v2.2.3
+ provider registry.terraform.io/hashicorp/local v2.3.0
+ provider registry.terraform.io/hashicorp/null v3.2.1
```

## Reproduction Code
Modify the build-package example's `package_dir_poetry` module below
https://github.com/terraform-aws-modules/terraform-aws-lambda/blob/1171025ea1e6d40bd7f37a63d130069b68ef8d5f/examples/build-package/main.tf#L49
by adding
```hcl
docker_additional_options = ["-e", "PIP_PLATFORM=manylinux2014_aarch64"]
```

- run the build-package example by following the readme
https://github.com/terraform-aws-modules/terraform-aws-lambda/tree/master/examples/build-package
- check the docker run command output.

## Expected behavior

docker_additional_options should be passed to the docker run command.

## Actual behavior

docker_additional_options is not passed to the docker run command.

### Terminal Output Screenshot(s)

<img width="1306" alt="image" src="https://user-images.githubusercontent.com/2734516/221501527-828da6d1-ad2c-4964-bb91-a7e5988f1f80.png">

## Additional context

The culprit is likely this line
https://github.com/terraform-aws-modules/terraform-aws-lambda/blob/1171025ea1e6d40bd7f37a63d130069b68ef8d5f/package.py#L1168
`docker` object is not passed to `docker_run_command`.

**Comment by pdecat** [comment]
Hi @FurqanHabibi,

you mention it is not working with poetry, does that mean it works with pip?

**Comment by antonbabenko** [comment]
This issue has been resolved in version 4.13.0 :tada:
